waiter/container_eacl.go

```diff
@@ -0,0 +1,86 @@
+package waiter
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+	apistatus "github.com/nspcc-dev/neofs-sdk-go/client/status"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	"github.com/nspcc-dev/neofs-sdk-go/eacl"
+)
+
+// ContainerSetEACLExecutor represents requirements to async container setEACL operation.
+// See documentation for functions in [client.Client]. The same semantics is expected.
+type ContainerSetEACLExecutor interface {
+	ContainerSetEACL(ctx context.Context, table eacl.Table, prm client.PrmContainerSetEACL) error
+	ContainerEACL(ctx context.Context, id cid.ID, prm client.PrmContainerEACL) (eacl.Table, error)
+}
+
+// ContainerSetEACLWaiter implements sync logic to container setEACL operation.
+type ContainerSetEACLWaiter struct {
+	executor     ContainerSetEACLExecutor
+	pollInterval time.Duration
+}
+
+// NewContainerSetEACLWaiter is a constructor for NewContainerSetEACLWaiter.
+func NewContainerSetEACLWaiter(c ContainerSetEACLExecutor, pollInterval time.Duration) ContainerSetEACLWaiter {
+	return ContainerSetEACLWaiter{executor: c, pollInterval: pollInterval}
+}
+
+// SetPollInterval allows rewrite default poll interval.
+func (w *ContainerSetEACLWaiter) SetPollInterval(interval time.Duration) {
```

Comment:
Do we need this setter now?

Reply:
Yes and no. For current implementation no. it may be useful for user, who wants to adjust this value in runtime (and this user use this waiter `waiter.ContainerSetEACLWaiter` separate from `waiter.Waiter`) between requests.

If my case is unreal, I don't mind to remove this setter